autocheese.lua

```diff
@@ -0,0 +1,155 @@
+--@module = true
+
+local ic = reqscript('idle-crafting')
+
+---make cheese using a specific barrel and workshop
+---@param barrel df.item
+---@param workshop df.building_workshopst
+---@return df.job
+function makeCheese(barrel, workshop)
+    ---@type df.job
+    local job = ic.make_job()
+    job.job_type = df.job_type.MakeCheese
+
+    local jitem = df.job_item:new()
+    jitem.quantity = 0
+    jitem.vector_id = df.job_item_vector_id.ANY_COOKABLE
+    jitem.flags1.unrotten = true
+    jitem.flags1.milk = true
+    job.job_items.elements:insert('#', jitem)
+
+    if not dfhack.job.attachJobItem(job, barrel, df.job_item_ref.T_role.Reagent, 0, -1) then
+        dfhack.error('could not attach item')
+    end
+
+    ic.assignToWorkshop(job, workshop)
+    return job
+end
+
+
+
+---unit is ready to take jobs
+---@param unit df.unit
+---@return boolean
+function unitIsAvailable(unit)
+    if unit.job.current_job then
+        return false
+    elseif #unit.individual_drills > 0 then
+        return false
+    elseif unit.flags1.caged or unit.flags1.chained then
+        return false
+    elseif unit.military.squad_id ~= -1 then
+        local squad = df.squad.find(unit.military.squad_id)
+        -- this lookup should never fail
+        ---@diagnostic disable-next-line: need-check-nil
+        return #squad.orders == 0 and squad.activity == -1
+    end
+    return true
+end
+
+---find unit with a particular labor enabled
+---@param unit_labor df.unit_labor
+---@param job_skill df.job_skill
+---@param workshop df.building
+---@return df.unit|nil
+---@return integer|nil
+ function findAvailableLaborer(unit_labor, job_skill, workshop)
+    local max_unit = nil
+    local max_skill = -1
+    for _, unit in ipairs(dfhack.units.getCitizens(true, false)) do
+        if
+            unit.status.labors[unit_labor] and
+            unitIsAvailable(unit) and
+            ic.canAccessWorkshop(unit, workshop)
+        then
+            local unit_skill = dfhack.units.getNominalSkill(unit, job_skill, true)
+            if unit_skill > max_skill then
+                max_unit = unit
+                max_skill = unit_skill
+            end
+        end
+    end
+    return max_unit, max_skill
+end
+
+local function findMilkBarrel(min_liquids)
```

Comment:
I don't see any logic to ensure that the milk barrel is reachable from the target workshop

Reply:
Good catch! I suppose the right fix is to only consider workshops that the selected (sufficiently full) barrel can be brought to. After all, the full barrel is the actual "trigger".